src/coreclr/interpreter/compiler.h

```diff
@@ -413,9 +414,54 @@ struct InterpVar
 
 struct StackInfo
 {
+private:
     StackType type;
+public:
+
     CORINFO_CLASS_HANDLE clsHnd;
 
+    StackType GetStackType()
+    {
+        if (type == StackTypeTransientPointer)
+        {
+            // Transient pointers are treated as byrefs for stack type purposes
+            return StackTypeByRef;
+        }
+        return type;
+    }
+
+    void SetAsTransientPointer()
+    {
+        assert(type == StackTypeByRef);
+        type = StackTypeTransientPointer;
+    }
+
+    bool IsTransientPointer()
+    {
+        return type == StackTypeTransientPointer;
+    }
+
+    // Used before a use of a value where the value on the stack would be correctly handled if the type on the stack
+    // was of type I. This is in support of section I.12.3.2.1 of ECMA-335
+    void BashStackTypeToI_ForTransientPointerUse()
```

Comment:
A nit - can we unify the naming style of `BashStackTypeToI_ForTransientPointerUse` and `BashStackTypeToIForConvert`?

Reply:
Done.